providers/ms365/resources/security_riskyusers.go

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) Mondoo, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package resources
+
+import (
+	"context"
+	"github.com/microsoftgraph/msgraph-sdk-go/identityprotection"
+	"github.com/microsoftgraph/msgraph-sdk-go/models"
+	"go.mondoo.com/cnquery/v11/llx"
+	"go.mondoo.com/cnquery/v11/providers-sdk/v1/plugin"
+	"go.mondoo.com/cnquery/v11/providers/ms365/connection"
+)
+
+// riskyUsers returns a list of risky users
+// requires IdentityRiskyUser.Read.All permission
+// see https://learn.microsoft.com/en-us/graph/api/resources/riskyuser?view=graph-rest-1.0
+func (a *mqlMicrosoftSecurity) riskyUsers() ([]interface{}, error) {
+	conn := a.MqlRuntime.Connection.(*connection.Ms365Connection)
+	graphClient, err := conn.GraphClient()
+	if err != nil {
+		return nil, err
+	}
+	ctx := context.Background()
+
+	filter := "riskState eq 'atRisk'"
+	resp, err := graphClient.IdentityProtection().RiskyUsers().Get(ctx, &identityprotection.RiskyUsersRequestBuilderGetRequestConfiguration{
```

Comment:
nit: is there any way we can check for the p2 license here so we can show a nice err msg or will the API itself throw such an err if the tenant is not p2?

Reply:
I think we need a better concept in general to check for permissions in Azure. Not just this one.